frontend/mr-admin-flate/src/components/portableText/PortableTextEditor.tsx

```diff
@@ -20,75 +23,146 @@ const HOTKEYS: {
   const editor = useMemo(() => withLinks(withHistory(withReact(createEditor()))), []);
-  const initialValue =
-    props.initialValue.length > 0
-      ? portableTextToSlate(props.initialValue)
-      : [
-          {
-            type: "paragraph",
-            children: [{ text: "" }],
-          },
-        ];
 
   return (
     <>
-      <Slate
-        editor={editor}
-        initialValue={initialValue}
-        onChange={(value) => {
-          if (value) {
-            props.onChange(slateToPortableText(value));
-          }
+      <Controller
+        name={label}
+        {...rest}
+        render={({ field: { onChange, value, name }, fieldState: { error } }) => {
+          return (
+            <div className={styles.container}>
+              <label
+                className={classnames(styles.label, {
+                  "navds-sr-only": hideLabel,
+                })}
+                style={{
+                  fontSize: size === "small" ? "16px" : "18px",
+                }}
+                htmlFor={name}
+              >
+                <b>{label}</b>
+              </label>
+              {description && (
+                <label
+                  className={classnames(styles.description, {
+                    "navds-sr-only": hideLabel,
+                  })}
+                  style={{
+                    fontSize: size === "small" ? "16px" : "18px",
+                  }}
+                >
+                  {description}
+                </label>
+              )}
+              <Slate
+                editor={editor}
+                initialValue={
+                  value && value.length > 0 ? portableTextToSlate(value) : emptySlateValue
+                }
+                onChange={(value) => {
+                  const portableText = slateToPortableText(value);
+                  onChange(portableText);
+                  providedOnChange?.({
+                    target: { value: portableText },
+                  });
+                }}
+              >
+                <div
+                  className={classnames(styles.editor_wrapper, {
+                    [styles.editor_wrapper__focusVisible]: focused,
+                  })}
+                >
+                  <Toolbar>
+                    <LinkButton />
+                    <MarkButton
+                      format="bold"
+                      icon={
+                        <div>
+                          <b>B</b> bold
+                        </div>
+                      }
+                    />
+                    <MarkButton
+                      format="italic"
+                      icon={
+                        <div>
+                          <em>I</em> italic
+                        </div>
+                      }
+                    />
+                    <BlockButton format="heading-one" icon={<div>Heading</div>} />
+                    <BlockButton format="bulleted-list" icon={<BulletListIcon />} />
```

Comment:
Har vi mulighet for å støtte til lenker her? Det tenker jeg vi bør ha

Reply:
Lenker fungerer (er en LinkButton rett over her) men de er litt rare. Hvis man skriver vg.no kommer man til .../vg.no. Må skrive https:// foran.

Reply:
Skjønner. Er det noen mulighet for å intercepte det som blir skrevet, så vi kan legge inn https:// foran det de skriver?

Reply:
Har lagt på validering nå :)